src/main/java/io/cdap/plugin/gcp/publisher/source/PubSubStructuredRecordConverter.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcp.publisher.source;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.format.StructuredRecordStringConverter;
+import io.cdap.plugin.format.avro.AvroToStructuredTransformer;
+import io.cdap.plugin.gcp.common.MappingException;
+import io.cdap.plugin.gcp.publisher.PubSubConstants;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.BinaryDecoder;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.HashMap;
+
+/**
+ * PubSubStructuredRecordConverter for converting a PubSubMessage to StructuredRecord.
+ */
+public class PubSubStructuredRecordConverter {
+
+  /**
+   * Converts a PubSubMessage into a StructuredRecord based on the specified schema.
+   * If no schema is specified, the default schema is used.
+   */
+  public static SerializableFunction<PubSubMessage, StructuredRecord> getFunction(GoogleSubscriberConfig config) {
+    return (pubSubMessage) -> {
+
+      Schema customMessageSchema = getCustomMessageSchema(config);
+      final Schema outputSchema = config.getSchema();
+      final String format = config.getFormat();
+
+      // Convert to a HashMap because com.google.api.client.util.ArrayMap is not serializable.
+      HashMap<String, String> hashMap = new HashMap<>();
```

Comment:
Same here. Try not to use just class names as variables/fields/method names. In this case `attributes` would be a god name. Nice names make it easier to read the code later

Reply:
Part of existing code, but made the change.